include/libopencm3/usb/hid.h

```diff
@@ -42,8 +42,30 @@ LGPL License Terms @ref lgpl_license
 
 #define USB_CLASS_HID	3
 
-#define USB_DT_HID	0x21
-#define USB_DT_REPORT	0x22
+#define USB_HID_SUBCLASS_NO 0
+#define USB_HID_SUBCLASS_BOOT_INTERFACE 1
+
+#define USB_HID_PROTOCOL_NONE 0
+#define USB_HID_PROTOCOL_KEYBOARD 1
+#define USB_HID_PROTOCOL_MOUSE 2
+
+#define USB_HID_DT_HID 0x21
+#define USB_HID_DT_REPORT 0x22
+#define USB_HID_DT_PHYSICAL 0x23
+// old names without USB_HID_ prefix
+#define USB_DT_HID USB_HID_DT_HID
+#define USB_DT_REPORT USB_HID_DT_REPORT
```

Comment:
no // comments please, and if possible, see if you can get an attribute on the old name that triggers a warning.

Reply:
I'm not sure how to expand a macro for an integer into `#warning "whatever"`, if that's what you meant, open to suggestions.

Personally I'd just remove these 2 lines, as the errors that will surface should be pretty obvious to amend. I can do the PR for libopencm3-examples with the necessary adjustments as well.